### PR TITLE
Fix errors due to pydicom 3.0

### PR DESCRIPTION
**Pull request by ckolbPTB**
Pydicom 3.0 was released which led to several errors due to deprecation, see e.g. https://github.com/pydicom/pydicom/pull/1567/files#diff-9b5a2e48457143cde31b1bbe1c7546a6733e577c306cfe582d56099851e12468

### CODE REVIEW

**Comment by ckolbPTB** [comment]
Currently all warnings are raised as errors in pytest (see https://github.com/PTB-MR/mrpro/blob/ca2f07b7030fe8ce3674f23207f974efcbc09fd3/pyproject.toml#L66)

This means that e.g DeprecationWarnings in other packages lead to errors. @fzimmermann89 @schuenke : Is this the behaviour we want, or should it only lead to errors for warning raised directly by our mrpro code?

**Comment by fzimmermann89** [comment]
DeprecationWarnings as errors makes sense IMHO.

It would be better if we could somehow set a depth for this, i.e. only if we call deprecated code it should error, not if one of our dependencies calls a deprecated function. But I think that might not be possible to detect..

**Comment by fzimmermann89** [comment]
(If  third party code throws a DeprecationWarning that we cannot fix, we could add an ignore to the toml filter warning setting for this deprecating warning)

**Comment by fzimmermann89** [comment]
@CodiumAI-Agent /review

File: src/mrpro/data/IHeader.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def from_dicom_list(cls, dicom_datasets: Sequence[Dataset]) -> Self:
             list of dataset objects containing the DICOM file.
         """
 
-        def get_item(dataset: Dataset, name: str | TagType):
+        def get_item(dataset: Dataset, name: TagType):
```
Comment (fzimmermann89) [inline src/mrpro/data/IHeader.py:65]: why did you remove the str here?

File: src/mrpro/data/IHeader.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def from_dicom_list(cls, dicom_datasets: Sequence[Dataset]) -> Self:
...
             """Get item with a given name or Tag from a pydicom dataset."""
             tag = Tag(name) if isinstance(name, str) else name  # find item via value name
```
Comment (fzimmermann89) [inline src/mrpro/data/IHeader.py]: but kept it here

**Review by fzimmermann89 — COMMENTED** [review]
why did you remove the str type hints?

File: src/mrpro/data/IHeader.py (hunk 1 above)
Comment (ckolbPTB) [reply src/mrpro/data/IHeader.py:65]: TagType is a union of different types already including str so it is not needed here

Comment (ckolbPTB) [reply src/mrpro/data/IHeader.py:65]: Well to be more accurate it used to be a union so str | TagType was redundant but ok and now it is a TagAlias: https://github.com/pydicom/pydicom/blob/b2073cccbd92d537dfb307c4d351a267e602069e/src/pydicom/tag.py#L37

File: src/mrpro/data/IHeader.py (hunk 2 above)
Comment (ckolbPTB) [reply src/mrpro/data/IHeader.py]: Probably this can be removed, yes

**Comment by fzimmermann89** [comment]
Got it, TagType includes str

...and you already answered the question :D sorry

**Review by fzimmermann89 — APPROVED** [review]

**Review by fzimmermann89 — APPROVED** [review]

**Pull request merged**